src/Illuminate/Auth/Middleware/EnsureEmailIsVerified.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Illuminate\Auth\Middleware;
+
+use Closure;
+use Illuminate\Support\Facades\Redirect;
+use Illuminate\Contracts\Auth\MustVerifyEmail;
+
+class EnsureEmailIsVerified
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return \Illuminate\Http\Response
+     */
+    public function handle($request, Closure $next)
+    {
+        if (! $request->user() ||
```

Comment:
Should a non-existing user be redirected to the same page as an existing, but unverified user?

Reply:
@36864 It makes sense to, otherwise an attacker knows whether an email is in the database and therefore a viable target if a non-existing user gets a different response to an existing-but-unverified user.